### ISSUE TITLE
Notification order in teams wrong

### ISSUE THREAD

**Issue by movedoa** [jenkinsci/office-365-connector-plugin#229]
When i start multiple projects in jenkins that get executed on after the other, the notifications get sent in the wrong order.
Lets say i start a build for project x two times (second execution in queued and is starting as soon as the first one is finished).

The messages in teams are in the following order.
Run 1 - Started
Run 2 - Started
Run 1 - Finished <- Wrong, should be displayed before Run2 - Started
Run 2 - Finished

We used slack before with the slack jenkins plugin and it didn't have this issue so i'm pretty sure this is a problem in the office-365-connector.

**Comment by damianszczepanik** [comment]
I need to have details about configuration to be able to reproduce the issue.

**Comment by movedoa** [comment]
We use the "Throttle Concurrent Builds Plug-in" to throttle the builds with the same category for all projects and max concurrent builds settings of 1. In the Office 365 Connector settings of the project i have set the url of the webhook and i have checked everything except "Notify Not Built".

I just noticed that sometimes the order of the messages is correct so it seems that the order gets mixed up when two projects send notifications in very short succession. Maybe this is a "race condition" when the first post takes a bit longer so the second one finishes first?

**Comment by damianszczepanik** [comment]
Agree. Time for post or time for processing at cloud might not be deterministic,

**Comment by movedoa** [comment]
> Agree. Time for post or time for processing at cloud might not be deterministic,

If this is something that is caused by external factors, maybe a configurable delay between messages could mitigate the issue. I would guess that 1 or 2 seconds pause between posts would fix almost all cases of this.
